tfhe/src/shortint/server_key/mod.rs

```diff
@@ -71,31 +73,46 @@ pub struct ServerKey {
 /// Returns whether it is possible to pack lhs and rhs into a unique
 /// ciphertext without exceeding the max storable value using the formula:
 /// `unique_ciphertext = (lhs * factor) + rhs`
-fn ciphertexts_can_be_packed_without_exceeding_space(
-    lhs: &Ciphertext,
-    rhs: &Ciphertext,
+fn ciphertexts_can_be_packed_without_exceeding_space<OpOrder: PBSOrderMarker>(
+    lhs: &CiphertextBase<OpOrder>,
+    rhs: &CiphertextBase<OpOrder>,
     factor: usize,
 ) -> bool {
     let final_degree = (lhs.degree.0 * factor) + rhs.degree.0;
     final_degree < lhs.carry_modulus.0 * lhs.message_modulus.0
 }
 
-pub struct Accumulator {
-    pub acc: GlweCiphertextOwned<u64>,
+#[derive(Clone, Debug, PartialEq)]
+#[must_use]
+pub struct Accumulator<C: Container<Element = u64>> {
+    pub acc: GlweCiphertext<C>,
     pub degree: Degree,
 }
 
-pub struct BivariateAccumulator {
+pub type AccumulatorOwned = Accumulator<Vec<u64>>;
+pub type AccumulatorMutView<'a> = Accumulator<&'a mut [u64]>;
+pub type AccumulatorView<'a> = Accumulator<&'a [u64]>;
+
+#[must_use]
+pub struct BivariateAccumulator<C: Container<Element = u64>> {
     // A bivariate accumulator is an univariate accumulator
     // where the message space is shared to encode
     // 2 values
-    pub acc: Accumulator,
+    pub acc: Accumulator<C>,
     // By how much we shift the lhs in the LUT
     pub ct_right_modulus: MessageModulus,
 }
 
-impl BivariateAccumulator {
-    pub fn is_bivariate_pbs_possible(&self, lhs: &Ciphertext, rhs: &Ciphertext) -> bool {
+pub type BivariateAccumulatorOwned = BivariateAccumulator<Vec<u64>>;
+pub type BivariateAccumulatorMutView<'a> = BivariateAccumulator<&'a mut [u64]>;
+pub type BivariateAccumulatorView<'a> = BivariateAccumulator<&'a [u64]>;
+
```

Comment:
since `keyswitch_programmable_bootstrap_bivariate` and co were renamed to `unchecked_apply_lookup_table_bivariate` which is way better, I think that the acumulattor name has to be changed.

`generate_accumulator` -> `generate_lut`
`Accumulator` -> `Lut` / `LookupTable`
`BivaraiteAccumulator` -> `BivariateLut` / `BivariateLookupTable`